built_in_tasks/target_tracking_task.py

```diff
@@ -0,0 +1,494 @@
+'''
+A generic target tracking task
+'''
+import numpy as np
+import time
+import os
+import math
+import traceback
+import random
+from collections import OrderedDict
+
+from riglib.experiment import traits, Sequence, FSMTable, StateTransitions
+from riglib.stereo_opengl import ik
+from riglib import plants
+
+from riglib.stereo_opengl.window import Window
+from .target_graphics import *
+
+## Plants
+# List of possible "plants" that a subject could control either during manual or brain control
+cursor = plants.CursorPlant()
+shoulder_anchor = np.array([2., 0., -15])
+chain_kwargs = dict(link_radii=.6, joint_radii=0.6, joint_colors=(181/256., 116/256., 96/256., 1), link_colors=(181/256., 116/256., 96/256., 1))
+chain_20_20_endpt = plants.EndptControlled2LArm(link_lengths=[20, 20], base_loc=shoulder_anchor, **chain_kwargs)
+chain_20_20 = plants.RobotArmGen2D(link_lengths=[20, 20], base_loc=shoulder_anchor, **chain_kwargs)
+
+plantlist = dict(
+    cursor=cursor,
+    chain_20_20=chain_20_20,
+    chain_20_20_endpt=chain_20_20_endpt)
+
+class TargetTracking(Sequence):
+    '''
+    This is a generic cued target tracking skeleton, to form as a common ancestor to the most
+    common type of motor control tracking tasks.
+    '''
+    status = dict(
+        wait = dict(start_trial="target"),
+        target = dict(success="reward", timeout="wait"),
+        reward = dict(reward_end="wait", stoppable=False, end_state=True)
+    )
+
+    # initial state
+    state = "wait"
+    frame_index = 0 # index in the frame in a trial
+    total_distance_error = 0 # Euclidian distance between cursor and target during each trial
+    tries = 0 # Helper variable to keep track of the number of failed attempts at a given trial.
+    trial_timed_out = False # check if the trial is finished
+    sequence_generators = []
+    plant_position = []
```

Comment:
this isn't the right place to put self.plant_position, it will grow and grow and never be reset until you turn off bmi3d, it should go in `__init__()` or `init()` so it gets called for every new instance of the task